syz-manager/html.go

```diff
@@ -226,6 +227,18 @@ func (mgr *Manager) httpCover(w http.ResponseWriter, r *http.Request) {
 
 func (mgr *Manager) httpCoverCover(w http.ResponseWriter, r *http.Request,
 	rg *cover.ReportGenerator, do func(io.Writer, []cover.Prog) error) {
+	ifFilter := func(cover []uint32) []uint32 { return cover }
+	if filter := r.FormValue("filter"); filter == "yes" && mgr.coverFilter != nil {
+		ifFilter = func(cover []uint32) (ret []uint32) {
+			pcs := coverToPCs(rg, cover)
```

Comment:
We do coverToPCs 2 times and then discard results of the first one, also check `pc`, but append `cover[i]`.
I think it can be made simpler by wrapping coverToPCs and then calling convert instead of coverToPCs below:
```
convert := coverToPCs
if r.FormValue("filter") != "" && mgr.coverFilter != nil {
	convert = (rg *cover.ReportGenerator, cover []uint32) (ret []uint64) {
		for _, pc := range coverToPCs(rg, cover) {
			if mgr.coverFilter[uint32(pc)] != 0 {
				ret = append(ret, pc)
			}
		}
		return
	}
}
```